packages/react-instantsearch-dom/src/hooks/useAnswers.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { instantSearchContext } from 'react-instantsearch-core';
+import { createConcurrentSafePromise } from '../lib/createConcurrentSafePromise';
+import { debounce } from '../lib/debounce';
+
+function hasReactHooks() {
+  // >= 16.8.0
+  const [major, minor] = React.version.split('.').map(Number);
+  return major >= 17 || (major === 16 && minor >= 8);
+}
+
+export default function useAnswers({
+  searchClient,
+  queryLanguages,
+  attributesForPrediction,
+  nbHits,
+  renderDebounceTime = 100,
+  searchDebounceTime = 100,
+  ...extraParameters
+}) {
+  if (!hasReactHooks()) {
+    throw new Error(
+      `\`Answers\` component and \`useAnswers\` hook require all React packages to be 16.8.0 or higher.`
+    );
+  }
+  const context = useContext(instantSearchContext);
+  const [query, setQuery] = useState();
+  const [index, setIndex] = useState();
+  const [isLoading, setIsLoading] = useState();
```

Comment:
How about a status state instead of a loading state?

Reply:
normally status state would be appropriate, but here we don't need any other state, so isLoading should be sufficient. If we need to know about other state, or need to introduce something like `isLoaded`, then we can convert it to status state.

Reply:
In the context of Answers I think it makes perfect sense to introduce multiple status that reflects the current state of the search. Answers is tricky in terms of UX because it's slower than regular Algolia indices. IMO it would make sense to also have a "stalled" state to allow better rendering of the component.

Example:

- Wait for input (`idle`): display nothing
- Start typing (`loading`): display nothing
- Network is slow (`stalled`): display a scaffold

With the current solution it's not trivial to do.

Possible TypeScript definition:

```ts
type AnswersStatus = 'idle' | 'loading' | 'stalled';
```

We also use a similar concept for the Voice widget I believe. If this is too much of a change compared to how React InstantSearch functions, that's fair! Since the widget is experimental we can still change this in a minor.

Reply:
It's a good idea to split it into `loading` and `stalled` so that they can display a scaffold only on `stalled`. Since we're updating the vanilla PR and the React one at the same time, it will slow down the process at the moment. I'd like to get these two released first, so that customers can start using, and then I'll take a look at how to introduce these statuses.